lib/src/component/error_boundary_mixins.dart

```diff
@@ -78,30 +141,174 @@ abstract class _$ErrorBoundaryStateMixin implements UiState {
 mixin ErrorBoundaryMixin<T extends ErrorBoundaryPropsMixin, S extends ErrorBoundaryStateMixin>
     on UiStatefulComponent2<T, S> {
   @override
-  Map get initialState => newState()..hasError = false;
+  Map get defaultProps => (newProps()
+    ..identicalErrorFrequencyTolerance = Duration(seconds: 5)
+  );
+
+  @override
+  Map get initialState => (newState()
+    ..hasError = false
+    ..showFallbackUIOnError = props.fallbackUIRenderer != null
+  );
 
   @mustCallSuper
   @override
-  Map getDerivedStateFromError(dynamic error) {
+  Map getDerivedStateFromError(/*Error||Exception*/dynamic error) {
     return newState()..hasError = true;
   }
 
   @mustCallSuper
   @override
-  void componentDidCatch(dynamic error, ReactErrorInfo info) {
+  void componentDidCatch(/*Error||Exception*/dynamic error, ReactErrorInfo info) {
     if (props.onComponentDidCatch != null) {
       props.onComponentDidCatch(error, info);
     }
+
+    _handleErrorInComponentTree(error, info);
+  }
+
+  @mustCallSuper
+  @override
+  void componentDidUpdate(Map prevProps, Map prevState, [dynamic snapshot]) {
+    // If the child is different, and the error boundary is currently in an error state,
+    // give the child a chance to remount itself and "recover" from the previous error.
+    if (state.hasError) {
+      final childThatCausedError = typedPropsFactory(prevProps).children.single;
+      if (childThatCausedError != props.children.single) {
+        reset();
+      }
+    }
   }
 
   @override
   render() {
-    if (state.hasError && props.fallbackUIRenderer != null) {
-      return props.fallbackUIRenderer();
+    if (state.hasError && state.showFallbackUIOnError) {
+      return (props.fallbackUIRenderer ?? _renderStringDomAfterUnrecoverableErrors)(_lastError, _lastErrorInfo);
     }
 
     return props.children;
   }
+
+  // TODO: Add PropTypes
```

Comment:
Not in this PR... I just noticed that `validateProps` had been removed without having an analogous propTypes replacement.